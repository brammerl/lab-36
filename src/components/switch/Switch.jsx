import React from 'react';

import Switch from '@material-ui/core/Switch';
import { useTheme, useDispatch } from '../../hooks/appContext'
const SwitchComponent = ({dispatch}) => {
  const handleChange = () => {
    dispatch({ type: 'TOGGLE_THEME'})
  }
  return (
    <div>
      <Switch
        onChange={handleChange}
        name="checkedA"
        inputProps={{ 'aria-label': 'secondary checkbox' }}
      />
    </div>
  );
}

export default SwitchComponent;