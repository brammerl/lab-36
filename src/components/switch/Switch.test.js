import React from 'react';
import { shallow } from 'enzyme';
import Switch from './Switch';

describe('Switch component', () => {
  it('renders Switch', () => {
    const wrapper = shallow(<Switch />);
    expect(wrapper).toMatchSnapshot();
  });
});
  