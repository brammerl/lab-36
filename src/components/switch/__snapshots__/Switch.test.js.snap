// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component renders Switch 1`] = `
<div>
  <WithStyles(ForwardRef(Switch))
    inputProps={
      Object {
        "aria-label": "secondary checkbox",
      }
    }
    name="checkedA"
    onChange={[Function]}
  />
</div>
`;
